apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts

```diff
@@ -0,0 +1,72 @@
+import { createSelector } from '@reduxjs/toolkit'
+import { DateTime } from 'luxon'
+import { t } from 'features/i18n/i18n'
+import { selectFilteredEvents } from 'features/vessels/activity/vessels-activity.selectors'
+import { PortVisitSubEvent } from 'types/activity'
+import { EventTypes } from '../../../../libs/api-types/src/events'
+
+export const selectFilteredEventsWithMainPortVisit = createSelector(
+  [selectFilteredEvents],
+  (events) =>
+    events.reduce((previous, event) => {
+      if (event.type !== EventTypes.Port) return [...previous, event]
+
+      const isPortExit = event.portVisitSubEvent === PortVisitSubEvent.Exit
+      const isPortEntry = event.portVisitSubEvent === PortVisitSubEvent.Entry
+      const id = event.id.split('-').shift()
+      const portEntry =
+        (isPortEntry && event) || events.find((ev) => ev.id === `${id}-${PortVisitSubEvent.Entry}`)
+      const portExit =
+        (isPortExit && event) || events.find((ev) => ev.id === `${id}-${PortVisitSubEvent.Exit}`)
+
+      const durationDiff = DateTime.fromMillis(portExit.end as number).diff(
```

Comment:
do we need this if we have the property durationHrs?

Reply:
also you already have a property durationDescription

Reply:
Same as above, given that the start property of the exit port visit was already overwritten the duration in this case is always 0, thus we have to calculate it.